### ISSUE TITLE
[Bug] Padded convolution - All zeros on GPU

### ISSUE THREAD

**Issue by michele-arrival** [openvinotoolkit/openvino#9039]
##### System information (version)
- OpenVINO=> 2021.4.752
- Operating System / Platform =>  Windows 64 Bit
- Compiler :grey_question: => from installer + python-3.7.11 - anaconda3 64 Bit
- Problem classification => Model Conversion :grey_question:

##### Detailed description
I have a super simple ONNX model generated from pytorch like below - A 1D convolution with a padding of 1. 
When i import it in OpenVino-python and i try to run inference + compare outputs i get all zeros when run on GPU, the correct result is instead inferred on CPU.

When instead the 

##### Steps to reproduce
Generate ONNX + save expected output for input, like so (or download the 3 files: [files.zip](https://github.com/openvinotoolkit/openvino/files/7660983/files.zip)):

```
import numpy as np
import torch
from torch import nn

module_pt = nn.Conv1d(in_channels=78,
                      out_channels=64,
                      kernel_size=3,
                      padding=1,
                      bias=False)

np.random.seed(1234)
data = np.random.random(size=(1, 78, 150))
data_t = torch.from_numpy(data).to(torch.float32)

ref_out = module_pt(data_t)

np.save("input.npy", data)
np.save("output.npy", ref_out.detach().cpu().numpy())
torch.onnx.export(module_pt, data_t, "model.onnx")
```

Run the following code.

In brief, this code loads the onnx network, loads the numpy array for the input tensor, runs inference and compares with the numpy array loaded as reference output. Repeat for device = "GPU"

```
import numpy as np
from openvino.inference_engine import IECore

ie = IECore()
net = ie.read_network(model="model.onnx")

sample_input = np.load("input.npy")
output_ref = np.load("output.npy")

input_blob = next(iter(net.input_info))
out_blob = next(iter(net.outputs))

for device in ["CPU", "GPU"]:
    exec_net = ie.load_network(network=net, device_name=device)

    output_test = exec_net.infer(inputs={input_blob: sample_input})[out_blob]

    np.testing.assert_allclose(output_test, output_ref, rtol=1e-03, atol=1e-03), (output_test, output_ref)

    print(device, "ok")
```

the test asserts because the output of the convolution on GPU is all zeros:

```
CPU ok

AssertionError:
Not equal to tolerance rtol=0.001, atol=0.001

Mismatched elements: 9579 / 9600 (99.8%)
Max absolute difference: 1.2720033
Max relative difference: 1.
 x: array([[[0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],...
 y: array([[[-0.64862 , -0.382263, -0.505221, ..., -0.626626, -0.548777,
         -0.2749  ],
        [-0.053266, -0.033243, -0.057755, ...,  0.344207, -0.247998,...
```

When there is no dilation the issue doesn't persist!

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by Iffa-Meah** [comment]
Your model's basic functionality seems fine since I'm able to run it through OpenVINO Benchmark App.
However, since it's a simple model, it's insufficient to be tested on the OpenVINO classification sample for a better view of the result.

<img width="494" alt="cpu" src="https://user-images.githubusercontent.com/79643476/145097118-f98fbc66-cca2-4f45-962f-aa5ba82e09a3.png">
<img width="494" alt="gpu" src="https://user-images.githubusercontent.com/79643476/145097152-44027cd4-68eb-48cb-8b8b-b00cba8d1406.png">

In your native onnx model, you'll need to save your model to GPU instead of CPU (**I believe you are using PyTorch**):
"np.save("output.npy", ref_out.detach().cpu().numpy())"
probably to: "np.save("output.npy", ref_out.detach().cuda().numpy())"

Note that the PyTorch only supports the usage and inference on NVIDIA GPU (Cuda).

In spite of that, since Openvino with its GPU Plugin enables Intel GPU utilization, this should be achievable.

We'll get back to you with more detailed info on how to do this in OpenVINO.

**Comment by michele-arrival** [comment]
Hi @Iffa-Meah and thanks for your answer

You can's save anything from cuda directly to numpy. I.e. `tensor.detach().cuda().numpy()` won't work. It needs to be `.cpu()` first, then `.numpy()`.

Regardless, the problem is not that the output is not the same as the one recorded in the `npy` file, rather that it's not the same as if run on CPU. 
In fact it's way simpler: on CPU it's giving me some non-zero result, whereas on GPU it's all zeros!

You can forget everything else, the question becomes `why is the GPU giving all zeros as results on this simple onnx model` ?

**Comment by michele-arrival** [comment]
Feel free to use this other little snippet, where i load a bin/xml model, and print the result to check that they are zeroes:

model: [model.zip](https://github.com/openvinotoolkit/openvino/files/7674610/model.zip)

```
import numpy as np
from openvino.inference_engine import IECore, __version__

np.set_printoptions(edgeitems=8, linewidth=300,
                    formatter=dict(float=lambda x: "% .3f" % x))

print(__version__)

ie = IECore()
net = ie.read_network(model="model.xml", weights="model.bin")

np.random.seed(1234)
sample_input = np.random.random((1, 32, 150))
input_blob = next(iter(net.input_info))
out_blob = next(iter(net.outputs))


for device in ["CPU", "GPU"]:
    exec_net = ie.load_network(network=net, device_name=device)

    output_test = exec_net.infer(inputs={input_blob: sample_input})[out_blob]

    print(device, output_test.flatten())
```

Output:
```
2021.4.2-3974-e2a469a3450-releases/2021/4
CPU [ 0.118 -0.144  0.087 -0.013  0.308  0.001  0.006 -0.106 ...  0.171  0.061  0.011  0.262  0.195  0.137 -0.000 -0.269]
GPU [ 0.000  0.000  0.000  0.000  0.000  0.000  0.000  0.000 ...  0.000  0.000  0.000  0.000  0.000  0.000  0.000  0.000]
```

**Comment by rahashim** [comment]
Hello michele-arrival,

I confirm that using your simple test model with one Conv1d, bias=False and inferencing on GPU produce all zeros. Changing the bias=True produce non-zero numbers but not close to the values produced by the CPU! Changing your model to use Conv2d or Conv3d works perfectly on both CPU and GPU and produce the same results. We will investigate this further and get back to you.

Thanks,
Raad

**Comment by michele-arrival** [comment]
Thanks @rahashim !

**Comment by rahashim** [comment]
@michele-arrival, The issue has been resolved and the fix has been merged into the master branch.
